x-pack/plugins/ml/public/application/notifications/components/notifications_list.tsx

```diff
@@ -56,8 +61,22 @@ export const NotificationsList: FC = () => {
   } = useMlKibana();
   const { displayErrorToast } = useToastNotificationService();
 
+  const [lastCheckedAt, setLastCheckedAt] = useStorage(ML_NOTIFICATIONS_LAST_CHECKED_AT);
+  const timeFilter = useTimefilter();
   const timeRange = useTimeRangeUpdates();
 
+  const [globalState] = useUrlState('_g');
+
+  useEffect(function setTimeRangeOnMount() {
+    if (globalState?.time || !lastCheckedAt) return;
+
+    timeFilter.setTime({
+      from: moment(lastCheckedAt).toISOString(),
```

Comment:
Would it be better to remember the last setting in the time picker rather than setting it to the time of the latest checked notification - for example I end up continually resetting to `Last 24 hours` when I switch back to the page.

Reply:
Maybe we could add an extra column to the table (8.6+) which indicates if a notification is 'new'? At that point it might be worth changing the behavior to preserve the time picker setting from the last visit to the page. Till then, this current behavior is a good way of showing the unread notifications.

Reply:
Actually, I implemented it like this initially but then decided to change to behaviour [here](https://github.com/elastic/kibana/pull/140944/commits/4b5725112ff308836768b1891fa624d923711382). 
Because imagine you have a time picker set to some time in the past (this can easily happen when you open anomaly explorer). And when you click the Notification link in the side nav, it's going to preserve this time range and it might be confusing for the user.

Reply:
Yes I wasn't thinking of using the time range from the global state, but stashing the last applied time range on the notifications page, where a common use case might be 'Last 1 hour' for example.